src/Toolbar.tsx

```diff
@@ -1,19 +1,451 @@
-import React, { useState } from 'react';
+/* eslint-disable no-multi-assign */
+/* eslint-disable prefer-destructuring */
+import React, { useRef, useState, useEffect } from 'react';
+import styled from '@emotion/styled';
+import { css } from '@emotion/react';
+import { BiBold, BiUnderline, BiItalic, BiLink } from 'react-icons/bi';
 
 interface ToolbarProps {
   containerRef: React.MutableRefObject<HTMLDivElement | null>;
 }
 
-const Toolbar = ({ containerRef }: ToolbarProps) => {
-  const openTitleMenu = () => {};
+function Toolbar({ containerRef }: ToolbarProps) {
+  const [toolbarPosition, setToolbarPosition] = useState<[number, number]>([0, 0]);
+  const [showToolbar, setShowToolbar] = useState<boolean>(false);
+  const toolbarRef = useRef<HTMLDivElement | null>(null);
 
-  const setBold = () => {};
+  useEffect(() => {
+    document.addEventListener('pointerup', (e) => {
+      const selection: Selection | null = window.getSelection();
 
-  const setUnderLine = () => {};
+      if (selection && selection.type === 'Caret') {
+        setShowToolbar(false);
+      }
 
-  const setItalic = () => {};
+      if (selection && selection.type === 'Range') {
+        const range = selection.getRangeAt(0);
+        if (containerRef.current && containerRef.current.contains(range.commonAncestorContainer)) {
+          calculateToolBarPosition();
+        } else {
+          setShowToolbar(false);
+        }
+      } else {
+        setShowToolbar(false);
+      }
+    });
+  }, [containerRef]);
 
-  return <div>Toolbar</div>;
-};
+  const calculateToolBarPosition = () => {
+    const selection: Selection | null = window.getSelection();
```

Comment:
```suggestion
    const selection = window.getSelection();
```
window.getSelection()이 타입을 가지고 있기 때문에 자연스럽게 그 타입이 배정되게 하면 됩니다~